docs/how-it-works/internals.md

```diff
@@ -0,0 +1,120 @@
+
+
+#### Bootstrapping GeoArrow
+
+For performance reasons, Lonboard is very focused on using GeoArrow throughout the pipeline in both Python and JavaScript. At the time when Lonboard was started, there were no user-friendly tools for converting data to GeoArrow in client-side JavaScript. By supporting only data from Python and only Parquet data, we could focus on making data conversions from GeoPandas to GeoArrow in Python efficient and stable. This enables GeoArrow to be "just an implementation detail" and out of any primary public APIs while also working to bootstrap a GeoArrow ecosystem in Python and JavaScript.
+
+#### Authentication
+
+Since all data is loaded via Python over the Jupyter-provided WebSocket, Lonboard doesn't need to implement any sort of authentication itself. The user is able to use whatever credentials they're familiar with, say for S3, to access data. Those credentials never need to be passed to the browser.
+
+In contrast, having the JavaScript side load data directly from a URL requires some way for users to specify authorization headers or API keys for access to private datasets.
+
+### Layer Creation
+
+With the exception of the `BitmapLayer`, no _direct_ deck.gl layers are used. deck.gl layers are used _exclusively_ through the [`@geoarrow/deck.gl-layers`](https://github.com/geoarrow/deck.gl-layers) glue library. This library connects GeoArrow data to the low-level binary data API of each deck.gl layer. **All** accessors are passed in directly as binary buffers.
+
+This means that in JavaScript the data _never_ leaves an Arrow binary representation. It's parsed from Parquet to Arrow in WebAssembly, the Arrow buffers are copied out of WebAssembly memory, but never parsed to JSON strings or JavaScript objects.
+
+### Accessors and data-driven rendering
+
+Loading data from Python also helps to simplify data-driven rendering.
+
+#### No function serialization
+
+If we can make the data movement between Python and JavaScript fast enough, we can serialize _computed rendering properties_ rather than the function to generate them. Lonboard serializes data, not functions.
+
+Any accessor used to style data is computed in Python, stored as a binary array, serialized to Parquet and sent to the frontend. Other than parsing the input Parquet buffers to Arrow, no data transformations or modifications are done on the JavaScript side.
+
+In Pydeck users had to pass something like
+
+```py
+getRadius="@@=properties.valuePerSqm"
+```
+
+which would be expanded into
+
+```js
+getRadius: (object) => object.properties.valuePerSqm
+```
+
+on the JavaScript side.
+
+I find this syntax to be horrible UX. The user is no longer writing Python, they need to learn a new DSL just for this.
```

Comment:
> And if the pydeck binary transport is used, the API is close to lonboard, just rather than passing in the column, the column name in the dataframe is referenced:
> 
> ```python
> pdk.Layer(
>     "ScatterplotLayer",
>     df,
>     get_position="position",
>     ...
> )
> ```

In this case, updating an accessor from Python would require sending the entire dataset from Python to JS again.